### PR TITLE
Fix converting of editable package

### PR DESCRIPTION
**Pull request by Czaki**
When I try to convert PartSeg to npe2 I found that there is a problem with the proper finding path to the module when the package is installed in editable mode.

Maybe it is somehow connected to changes to setuptools mentioned by @tlambert03 in the past.

### CODE REVIEW

File: src/npe2/__main__.py
Comment (Czaki) [inline src/npe2/__main__.py:1]: I have added this file to simplify debug my allow run by `python -m`

File: src/npe2/_inspection/_from_npe1.py [hunk 1]
```diff
@@ -517,7 +517,13 @@ def get_top_module_path(package_name, top_module: Optional[str] = None) -> Path:
         top_module = top_mods[0]
 
     path = Path(dist.locate_file(top_module))
-    assert path.is_dir()
+    if not path.is_dir() and dist.files:
+        for f_path in dist.files:
+            if "__editable__" in f_path.name:
+                path = Path(f_path.read_text().strip()) / top_module
+                break
```
Comment (jni) [inline src/npe2/_inspection/_from_npe1.py:524]: @Czaki I have zero context here so take my review with a grain of salt, but if there are no tests here, could you at least add comments about why these lines are needed? eg "if the given npe1 plugin is installed in editable mode, then X doesn't work so we must do Y"

**Comment by tlambert03** [comment]
Yes @Czaki, this is the weird editable mode used by setuptools for a pyproject only package

**Comment by Czaki** [comment]
@tlambert03 partseg has `setup.py` (concat changelog with readme) , `setup.cfg`, and `pyproject.toml` files. So it looks like it is enough to contain `pyproject.toml`. And it looks like the problem surface on the drop of npe1 may be big without handling this. 

How did you think? Should I implement some simple package to write a test for it, or just add comments?

**Comment by tlambert03** [comment]
Would be great to test it. I can't remember, did I make a test case for a setup.cfg repo here?  If so, could you use it as a template to copy?

**Comment by Czaki** [comment]
@tlambert03 @jni I have finally created the proper test.

**Review by tlambert03 — APPROVED** [review]

**Pull request merged**